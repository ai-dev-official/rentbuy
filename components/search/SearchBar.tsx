import React from 'react';
import { StyleSheet, Text, View, ViewStyle, TextStyle } from 'react-native';
import { useTheme } from 'react-native-paper';
const SearchIcon = require('react-native-vector-icons/Feather').default;
const DotIcon = require('react-native-vector-icons/Octicons').default;



// Define the type for styles
type Styles = {
  wrapper: ViewStyle;
  trailingText: ViewStyle;
  search?: ViewStyle; // Optional styles if you plan to add more properties to search icon
  searchTexts?: ViewStyle; // Optional style if you plan to add styles to searchTexts View
  leadingText?: TextStyle; // Optional style if you plan to add styles to leadingText
  dot?: ViewStyle; // Optional styles for DotIcon
  placeholderText: TextStyle;
};

const SearchBar: React.FC = () => {
  const { colors } = useTheme();

  const styles = StyleSheet.create<Styles>({
    wrapper: {
      borderWidth: 1,
      borderRadius: 20,
      paddingHorizontal: 20,
      height: 60,
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center',
      borderColor: colors.primary, // Optional: using theme color
    },
    search:{
      paddingRight: 20,
    },
    trailingText: {
      flexDirection: 'row',
      alignItems: 'center', // Changed from alignContent to alignItems
      justifyContent: 'space-between',
    },
    searchTexts:{
      justifyContent: 'space-evenly',
    },
    leadingText:{
      fontSize: 18,
      fontWeight: '700',
    },
    dot:{
      marginHorizontal: 20,
    },
    placeholderText:{
      color: 'grey',
    }
  });

  return (
    <View style={styles.wrapper}>
      <SearchIcon name="search" size={36} style={styles.search} />

      <View style={styles.searchTexts}>
        <Text style={styles.leadingText}>Where to?</Text>
        <View style={styles.trailingText}>
          <Text style={styles.placeholderText}>Anywhere</Text>
          <DotIcon name="dot-fill" size={12} style={styles.dot} />
          <Text style={styles.placeholderText}>Any week</Text>
          <DotIcon name="dot-fill" size={12} style={styles.dot} />
          <Text style={styles.placeholderText}>Add guests</Text>
        </View>
      </View>
    </View>
  );
};

export default SearchBar;
