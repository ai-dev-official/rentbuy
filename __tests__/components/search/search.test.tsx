import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../../../components/search/SearchBar';
import {it} from '@jest/globals';
import { Text, View } from 'react-native';


describe('SearchBar Component', () => {
  // Snapshot test to ensure UI rendering
  it('renders correctly', () => {
      const tree = renderer.create(<SearchBar />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  // Test if the SearchBar component contains the correct number of text elements
  it('should render the correct number of text elements', () => {
      const instance = renderer.create(<SearchBar />);
      const textInstances = instance.root.findAllByType(Text);
      expect(textInstances.length).toBe(7); // Update to reflect the actual count: "Where to?", "Anywhere", "Any week", "Add guests", and 3 other text components (like separators).
  });

  

// Test if the leading text is rendered correctly

it('should render leading text correctly', () => {
  const instance = renderer.create(<SearchBar />);
  
  // Find all text instances
  const textInstances = instance.root.findAllByType(Text);

  // Log the text instances to inspect their structure
  console.log(textInstances.map(text => text.props.children));

  // Look for the specific text content within the text instances
  const leadingText = textInstances.find(text => text.props.children === 'Where to?');

  expect(leadingText).toBeTruthy(); // Check if the leading text exists
  expect(leadingText?.props.children).toBe('Where to?'); // Verify the content of the leading text
});




  // Test if the search icon is rendered
  it('should render the search icon', () => {
      const instance = renderer.create(<SearchBar />);
      const searchIcon = instance.root.findByProps({ name: "search" });
      expect(searchIcon).toBeTruthy();
  });

  // Test if the dot icons are rendered
  it('should render dot icons correctly', () => {
      const instance = renderer.create(<SearchBar />);
      const dotIcons = instance.root.findAllByProps({ name: "dot-fill" });
      expect(dotIcons.length).toBe(2); // There should be two dot icons
  });

  // Test if the search icon has the correct size
  it('should have the search icon size of 36', () => {
      const instance = renderer.create(<SearchBar />);
      const searchIcon = instance.root.findByProps({ name: "search" });
      expect(searchIcon.props.size).toBe(36);
  });

  // Test if the styles are applied correctly
  it('should apply correct styles to the wrapper', () => {
      const instance = renderer.create(<SearchBar />);
      const wrapper = instance.root.findByType(View);
      expect(wrapper.props.style).toEqual(
          expect.objectContaining({
              borderWidth: 1,
              borderRadius: 20,
              height: 60,
              flexDirection: 'row',
              justifyContent: 'flex-start',
              alignItems: 'center',
          })
      );
  });
});