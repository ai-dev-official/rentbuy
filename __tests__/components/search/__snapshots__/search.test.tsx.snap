// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "rgba(103, 80, 164, 1)",
      "borderRadius": 20,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "flex-start",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 36,
        },
        {
          "paddingRight": 20,
        },
        {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      {
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
      Where to?
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "color": "grey",
          }
        }
      >
        Anywhere
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "marginHorizontal": 20,
            },
            {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "color": "grey",
          }
        }
      >
        Any week
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "marginHorizontal": 20,
            },
            {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "color": "grey",
          }
        }
      >
        Add guests
      </Text>
    </View>
  </View>
</View>
`;
